web/webpack.config.js

```diff
@@ -59,7 +61,37 @@ module.exports = function(env) {
                 minify: true,
                 navigateFallback: "/index.html"
             }),
-            new webpack.HotModuleReplacementPlugin()
+            new webpack.HotModuleReplacementPlugin(),
+            new HtmlWebpackPlugin({
+				filename: "index.html",
+				template: path.resolve(__dirname, "src/index.html"),
+				inject: true,
+				minify: {
+					removeComments: env === "prod",
+					collapseWhitespace: env === "prod",
+					removeAttributeQuotes: env === "prod"
+				},
+            }),
+			new FriendlyErrors(),
+			new webpack.optimize.CommonsChunkPlugin({
+				name: "common",
+				minChunks: function (module, count) {
+					// any required modules inside node_modules are extracted to vendor
+					return (
+						module.resource &&
+						/\.js$/.test(module.resource) &&
+						module.resource.indexOf(
+							path.join(__dirname, "../node_modules")
+						) === 0
+					);
+				}
+			}),
+			// extract webpack runtime and module manifest to its own file in order to
+			// prevent commons hash from being updated whenever app bundle is updated
+			new webpack.optimize.CommonsChunkPlugin({
+				name: "common",
+				chunks: ["common"]
+			})
         ]
```

Comment:
Get rid of the tabs here and I'll merge 👍

Reply:
Offtop, config should probably be separated using [webpack-merge](https://github.com/survivejs/webpack-merge), but when I tried to do that webpack build failed with a weird reason. Gonna try to investigate further.